src/x11/x.c

```diff
@@ -161,8 +161,24 @@ static bool x_win_composited(struct window_x11 *win)
 void x_display_surface(cairo_surface_t *srf, window winptr, const struct dimensions *dim)
 {
         struct window_x11 *win = (struct window_x11*)winptr;
-        x_win_move(win, dim->x, dim->y, dim->w, dim->h);
-        cairo_xlib_surface_set_size(win->root_surface, dim->w, dim->h);
+        const struct screen_info *scr = get_active_screen();
+        int scale = x_get_scale();
```

Comment:
Could you pass the scale as a variable to the function instead? It is known in `void draw()` already

Reply:
It's also called from `x_win_show()`.

Reply:
Oh it's fine then.